### ISSUE TITLE
Rails 5.0.0.1 --api, Ember 2.7, ember-cli-rails does not mount the ember app

### ISSUE THREAD

**Issue by rmcsharry** [thoughtbot/ember-cli-rails#480]
**Which operating system and version is the project developed on?**
OSX El Capitan

**Which version of [`ruby`][ruby] is the project developed on?**
2.3.1

**Which version of [`npm`][npm] is the project developed on?**
3.10.6

**Which version of [`ember-cli`][ember-cli] is the project developed on?**
2.7

**What is the [`rails`][rails] version?**
5.0.0.1

**What is the [`ember-cli-rails`][gem] version (from `Gemfile`)?**
0.8.0

**What is the [`ember-cli-rails-addon`][addon] version (from `package.json`)?**
0.8.0

**Is your application server [multi-threaded][] (such as `puma` and `unicorn`) or is
it multi-process (such as thin and webrick)?**
puma

**What are the contents of `config/initializers/ember.rb`?**
EmberCli.configure do |c|
  c.app :gearclient, path: "frontend"
end

**What are the contents of the Rails' view that renders the Ember application?**
It's empty - the guide says nothing about it (unless you want to override it). By default the EmbercliController should render the ember app's index.html, but it doesn't. Seems it cannot find or mount the ember app. 

I tried putting an empty view here:
app/views/ember_cli/index.html.erb

Also tried filling it with:
<%= render_ember_app ember_app do |head| %>
  <% head.append do %>
    <%= csrf_meta_tags %>
  <% end %>
<% end %>

But neither options worked.

**How are the EmberCLI-related routes defined?**
mount_ember_app :gearclient, to: "/"

**How is the application deployed?**
It's not, this is locally in development

In Rails root is a folder called frontend. Inside that is the full ember app called 'gearclient' (ie. it is not in a subfolder of frontend, but the ember app in package.json is called 'gearclient')

[ruby]: https://www.ruby-lang.org/
[node]: https://nodejs.org/en/
[npm]: https://www.npmjs.com/
[ember-cli]: http://ember-cli.com/
[rails]: https://github.com/rails/rails
[gem]: https://github.com/thoughtbot/ember-cli-rails
[addon]: https://github.com/rondale-sc/ember-cli-rails-addon/
[multi-threaded]: https://github.com/thoughtbot/ember-cli-rails/commit/ef0a49546a8c2e5fb0a298ba455fba97289065a1

The output from Rails log is this, clearly shows that the Ember App is not bootstrapped/mounted.

=> Booting Puma
=> Rails 5.0.0.1 application starting in development on http://localhost:3000
=> Run `rails server -h` for more startup options
Puma starting in single mode...
* Version 3.6.0 (ruby 2.3.1-p112), codename: Sleepy Sunday Serenity
* Min threads: 5, max threads: 5
* Environment: development
* Listening on tcp://localhost:3000
Use Ctrl-C to stop
I, [2016-08-24T23:58:38.637967 #41788]  INFO -- : Started GET "/" for ::1 at 2016-08-24 23:58:38 +0200
D, [2016-08-24T23:58:39.392008 #41788] DEBUG -- :   ActiveRecord::SchemaMigration Load (1.2ms)  SELECT "schema_migrations".* FROM "schema_migrations"
I, [2016-08-24T23:58:39.414124 #41788]  INFO -- : Processing by EmberCli::EmberController#index as HTML
I, [2016-08-24T23:58:39.414219 #41788]  INFO -- :   Parameters: {"ember_app"=>:frontend}
I, [2016-08-24T23:58:39.415062 #41788]  INFO -- : Completed 200 OK in 1ms (Views: 0.1ms | ActiveRecord: 0.0ms)

I have also tried changing the initializer and the route to use :frontend instead of :gearclient, but also did not work.

**Comment by seanpdoyle** [comment]
@rmcsharry thanks for opening this issue.

Have you compared your approach to the one illustrated in the [example repository](https://github.com/seanpdoyle/ember-cli-rails-heroku-example)?

Additionally, this gem is tested against [a dummy Rails 5 application](https://github.com/thoughtbot/ember-cli-rails/tree/19361c1fd9b84de92c1b043d861d5a6b6b3703b7/spec).

Could you compare your configuration against those working examples? If you find anything that you're doing differently that might help us understand the root of your issue.

**Comment by rmcsharry** [comment]
@seanpdoyle Thanks for the fast response. Yes I've compared it, the only difference I see now is the '2nd app' in the example has an AdminController (ie named after the app). So I will try adding that and let you know.

However, I also reset the code and tried again with a brand new ember app (ember new frontend) inside folder 'frontend' as per the example. Rails did not build this either.

Could it be something to do with using Rails new --api flag when I first setup up Rails? Is the dummy app a Rails 5 api app or a full Rails app?

**Comment by rmcsharry** [comment]
@seanpdoyle [This]
(https://github.com/rmcsharry/ertest) vanilla repro reproduces the problem.

1 rails new ertest --api
2 add ember-cli-rails to gemfile
3 bundle install  (installs 0.8.0)
4 rails g ember:init
5 config/routes.rb -> add mount_ember_app :frontend, to: "/" 
6 ember new frontend --git-skip
7 cd frontend
8 ember install ember-cli-rails-addon (installs 0.8.0)
9 added 'test' to frontend/app/index.html
10 rails server

Same problem - 'test' (from step 9) is not rendered as it seems rails does not mount the frontend app.

**Comment by seanpdoyle** [comment]
@rmcsharry thanks a lot for those details, and *especially* a public repository that reproduces your issue!

**Comment by seanpdoyle** [comment]
@rmcsharry I've created https://github.com/thoughtbot/ember-cli-rails/pull/481 and tested it against your sample application.

It doesn't yet resolve your issue, but it's a good start. 

The next step is to add feature-level test coverage of `ember-cli-rails` running against a `rails new --api` application.

**Comment by rmcsharry** [comment]
@seanpdoyle A great start, and so fast too!

It's nice to know my hunch was right about using `rails new --api` 👍 

I wish I had the expertise (and time) to assist you but I'm afraid I have neither. :(

The main reason I chose this gem was to make life easier (eg. fast deploys to Heroku), rather than having to mess with deploying two separate apps. I'd rather use the gem if I can, so just wondering if you have a rough estimate on how long you think it might take to fix?

**Comment by seanpdoyle** [comment]
In the mean time, could you try out [PR#481](https://github.com/thoughtbot/ember-cli-rails/pull/481), and let us know if that resolves your issue:

```rb
# Gemfile

gem "ember-cli-rails", github: "thoughtbot/ember-cli-rails", branch: "support-api-controller"
```

**Comment by rmcsharry** [comment]
@seanpdoyle Yes I'll do that tonight and let you know.

**Comment by rmcsharry** [comment]
@seanpdoyle Awesome! The ertest app works with that branch 👍 

I will try it on my actual app now and report back.

**Comment by rmcsharry** [comment]
It works. :) You are my new hero!

**Comment by seanpdoyle** [comment]
Published in [0.8.0](https://github.com/thoughtbot/ember-cli-rails/releases/tag/v0.8.1)
